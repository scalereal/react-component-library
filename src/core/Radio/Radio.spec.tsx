import { render } from '@testing-library/react';
import * as React from 'react';
import { Radio } from '.';

describe('Radio component', () => {
    it('should render with default theme', () => {
        const { container } = render(<Radio/>);
        expect(container).toMatchSnapshot();
    }); 
    
    it('should render with all the props given', () => {
        const { container } = render(
            <Radio
                label="1"
                value={1}
                size="M"
                labelColor='grey'
                bgColor= '#67686b'
                hoverCheckedBgColor='#126AFA'
                checkedBgColor='#126AFA'
                hoverEffect={true}
                labelHoverEffect={true}
                hoverBgColor='#373738'
                hoverLabelColor= '#12fadb'
            />
            );
        expect(container).toMatchSnapshot();
    });

    it('should render with not-allowed cursor when it is disabled', () => {
        const { container } = render(
            <Radio disabled={true} />
        );
        expect(container).toMatchSnapshot();
    });




});