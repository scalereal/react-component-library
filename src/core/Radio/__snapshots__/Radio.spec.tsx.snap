// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component should render with all the props given 1`] = `
<div>
  <div
    class="sc-eCssSg flCDJa"
  >
    <label
      class="sc-dlfnbm cayrCo"
      for="error"
      id="error"
    />
    <label
      class="sc-hKgILt firMIG"
      label="1"
      value="1"
    >
      <span
        class="sc-bdfBwQ fVhTiG radioLabel"
      >
        1
      </span>
      <input
        type="radio"
        value="1"
      />
      <span
        class="sc-gsTCUz uHtaa"
        label="1"
        value="1"
      />
    </label>
  </div>
</div>
`;

exports[`Radio component should render with default theme 1`] = `
<div>
  <div
    class="sc-eCssSg flCDJa"
  >
    <label
      class="sc-dlfnbm cayrCo"
      for="error"
      id="error"
    />
    <label
      class="sc-hKgILt dFxNXU"
    >
      <span
        class="sc-bdfBwQ cmSgzP radioLabel"
      >
        CheckBox
      </span>
      <input
        type="radio"
        value=""
      />
      <span
        class="sc-gsTCUz jcBRty"
      />
    </label>
  </div>
</div>
`;

exports[`Radio component should render with not-allowed cursor when it is disabled 1`] = `
<div>
  <div
    class="sc-eCssSg flCDJa"
  >
    <label
      class="sc-dlfnbm cayrCo"
      for="error"
      id="error"
    />
    <label
      class="sc-hKgILt rqhIk"
      disabled=""
    >
      <span
        class="sc-bdfBwQ cmSgzP radioLabel"
      >
        CheckBox
      </span>
      <input
        type="radio"
        value=""
      />
      <span
        class="sc-gsTCUz jcBRty"
        disabled=""
      />
    </label>
  </div>
</div>
`;
